### PR TITLE
[Console] Preserving line breaks between sentences according to the exception message

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #24212 
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Review by ro0NL — APPROVED** [review]
:+1: for preserving lines. Think intention was pure for cosmetic reasons.

File: src/Symfony/Component/Console/Tests/ApplicationTest.php [hunk 1]
```diff
@@ -608,6 +608,22 @@ public function testRenderExceptionEscapesLines()
...
+        $application->setAutoExit(false);
+        $application->expects($this->any())
+            ->method('getTerminalWidth')
+            ->will($this->returnValue(120));
+        $application->register('foo')->setCode(function () {
+            throw new \InvalidArgumentException("Command \"foO\" is not defined.\n\nDid you mean this?\n    foo");
+        });
+        $tester = new ApplicationTester($application);
+
+        $tester->run(array('command' => 'foo'), array('decorated' => false));
```
Comment (ro0NL) [inline src/Symfony/Component/Console/Tests/ApplicationTest.php]: just be sure.. decorated=true still works as expected?

File: src/Symfony/Component/Console/Application.php [hunk 2]
```diff
@@ -1133,9 +1133,8 @@ private function splitStringByWidth($string, $width)
             $lines[] = str_pad($line, $width);
             $line = $char;
         }
-        if ('' !== $line) {
```
Comment (ro0NL) [inline src/Symfony/Component/Console/Application.php]: @yceruto i think this is trimming related in fact :)

```
throw new \Exception("\n\nline 1 with extra spaces        \nline 2\n\nline 4\n");
```

Before/After
image

image

So here's a better test ;-)

Comment (yceruto) [reply src/Symfony/Component/Console/Application.php]: Done.

Comment (ro0NL) [reply src/Symfony/Component/Console/Application.php]: I think the trimming should be preserved ;-)

**Review by chalasr — APPROVED** [review]

**Review by ogizanagi — APPROVED** [review]

**Comment by chalasr** [comment]
Thanks for fixing this bug Yonel.

**Pull request merged**